### ISSUE TITLE
Reloading is not working for changes inside folder

### ISSUE THREAD

**Issue by bruno02221** [parcel-bundler/parcel#382]
🐛 bug report

I have a very simple scenario: My project folder has only an `index.html` in the root and an `index.js` in the `src` folder. I run the development server with: `parcel index.html`
Then when I make changes in the `src/index.js` file the page doesn't reload its content. Even if I refresh my browser manually the content doesn't change.
I've tested the same scenario but with `index.js` in the root folder, and it works perfectly.

### 🎛 Configuration (.babelrc, package.json, cli command)

package.json
```js
{
  "name": "parcel-test",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "keywords": [],
  "author": "",
  "license": "ISC"
}
```

### 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | 1.3.0
| Node             | v8.9.1
| npm/Yarn         | npm 5.6.0
| Operating System | Ubuntu 16.04

**Comment by mrolla** [comment]
Can you show your `index.html`? I suspect it's pointing to the wrong `index.js`.

**Comment by bruno02221** [comment]
```html
<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <title></title>
</head>
<body>
  <script src="./src/index.js"></script>
</body>
</html>
```
I've already tested using `<script src="src/index.js"></script>` and it gets the same problem. It initially gets the right content, but it do not update when I change the `index.js` file.

**Comment by mrolla** [comment]
That's weird, since this is exactly how I'm using parcel. Do you mind creating a minimum repo reproducing the issue so I can take a look at it?

**Comment by shawwn** [comment]
Try running `parcel --no-cache index.html`. Does that fix the issue?

**Comment by bruno02221** [comment]
Steps to reproduce:

```bash
mkdir my-app
cd my-app
mkdir src
touch src/index.js
touch index.html
```

index.html
```html
<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <title></title>
</head>
<body>
  <script src="./src/index.js"></script>
</body>
</html>
```

src/index.js
```js
document.body.innerHTML = "Initial content";
```

`parcel index.html`

It renders correctly in the first time.
But when I make changes in the `src/index.js` file, nothing happens. I can even
hard refresh the browser and it gets the same content as before.

I tried with:
`parcel --no-cache index.html` and it still the same.

**Comment by shawwn** [comment]
Interesting. Parcel v1.1.0, the latest HEAD version, and one from a few
commits ago all successfully let you change the JS file and perform hot
reloading in Chrome. I performed your repro steps exactly.

Try `git clone https://github.com/parcel-bundler/parcel && cd parcel &&
yarn`, then run bin/cli.js on your index.html file. (parcel is simply an
alias pointing to bin/cli.js)

If the latest HEAD version doesn't work, this is very interesting. Which
browser are you using?

**Comment by bruno02221** [comment]
I'm using Google Chrome 63.0.3239.108

Man it looks like I'm doing something very stupid but I can't figure out what it is. But I'm going to try with the latest HEAD to check it out.

**Comment by shawwn** [comment]
Don't worry, there are no stupid bug reports. Whatever you're experiencing,
some other user will experience it too. It's useful for us to either
provide a google trail for them to follow (here) or update the
documentation.

**Comment by bruno02221** [comment]
I tried with the latest HEAD and it's still not updating the content after changes.

**Comment by shawwn** [comment]
Now you've piqued my interest. This mystery must be solved.

Feel like doing some debugging? I have to go see family for a few hours,
but if you throw in some `console.log("foo")` statements at strategic
points in the codebase, you should be able to work out why it's not
reloading.

I suggest throwing a console log into Bundler.onChange(), which will tell
you whether the FS is properly seeing filesystem change events whenever you
update src/index.js.

In fact, that is the most reasonable explanation: Our FS watcher doesn't
work on Ubuntu. If so, then this was a very useful exercise.

If that turns out to be correct, Can you provide more detail about where
you're storing the files? What happens if you try all of this in /tmp? I'm
wondering if some symlinks are to blame.

**Comment by DeMoorJasper** [comment]
Perhaps see #221 , this probably is safe-write

**Comment by bruno02221** [comment]
@DeMoorJasper that's exactly the same problem I was having. I'm using Vim and after I changed my `.vimrc` file to include these lines:

```
set nobackup
set nowritebackup
```

the problem is solved. Thank you guys.

**Comment by boazblake** [comment]
works on PC but not on a MAC

**Comment by mindplay-dk** [comment]
Every time I use Parcel, I need to dig up this issue and solve the problem again.

Why do I have this problem with Parcel only?

Never had this problem with e.g. webpack or rollup.

It seems a bit problematic that failure is the default for people working with (at least) any of the JetBrains IDEs or VS Code on PC, probably others. 🤔

**Comment by viderme** [comment]
@mindplay-dk , how did you solve this problem on vscode?

**Comment by GeorgeFlorian** [comment]
I have the exact same problem in VSC in Ubuntu 20.04.
Is there any fix ?

**Comment by ShankhadeepBanerjee** [comment]
I removed the 

`{
"main": "index.js",
}`

in package.json
now it is working.


before this I had to press `CTRL + SHIFT + R` to see changes in the webpage.

**Comment by erckmartn** [comment]
I'm using Parcel and I also have this problem and no matter how much I modified my .js file it didn't modify in the browser.
I used CRTL +SHIFT + R. Which forcefully reloads the page. It worked, but it is annoying to use it every time.

The solution I found. 
It is in the official Parcel documentation.
[https://parceljs.org/features/development/](url)
It talks about the cache folder, which is created by parcel, to not use it you just have to append `--no-cache` inside the execution of your yarn or npm command. 

I modified my 'scripts' inside my package.json file by inserting the `--no-cache` .

`"scripts": {
    "start": "parcel --no-cache",
    "build": "parcel build"
  }`

Now to start my development server I just use yarn start 

I hope this is helpful. 
Greetings, good luck with your developments friends.
